third_party/amd/lib/TritonAMDGPUTransforms/OptimizeAtomicLayouts.cpp

```diff
@@ -0,0 +1,140 @@
+#include "TritonAMDGPUToLLVM/TargetUtils.h"
```

Comment:
would it make sense to put this logic in the Coalesce pass?

Reply:
I'm completely Ok with this idea, but I'm not sure when this logic makes sense for nvidia targets, need to investigate it.

Also I just found an alternative solution, which is a bit better for AMD, so turning PR to DRAFT for now.
If this functionality is good for nvidia, lets continue working on it.

Reply:
Didn’t look too closely at your PR, but my understanding is that you need the input to the atomic operation to be in a specific layout.

This is also true for nvidia, but the specific layouts needed might be different.

Reply:
@joviliast curious,  can you link where you did the alternative approach for AMD?

Reply:
https://github.com/triton-lang/triton/pull/5072 .
I did not find the case where current PR is better then dpp approach.